feat: open goal modal with spacebar key

harshalranjhani opened the pull request:
the goal modal now opens with spacebar key, a similar action which could be performed by clicking the goal as well

fixes #2053 

Screenshot:
![image](https://github.com/user-attachments/assets/d31b5bf7-7435-4d7b-8929-e7816c6e17ab)

tijlleenders commented:
@harshalranjhani The original intent of the issue was to navigate 'into' the goal - instead of opening the modal...   
... but actually the way you naturally interpreted it probable makes more sense. Keep the left/right/up/down keys for navigation, and spacebar for 'action'.  


I'm good with it - except that the background navigates to a higher lever when one presses spacebar at a lower level:  
[Screencast from 2024-10-13 14-09-20.webm](https://github.com/user-attachments/assets/0a95e5f2-546a-478a-a3d5-3328760843ec)

harshalranjhani commented:
> background navigates to a higher lever when one presses spacebar at a lower level

hii, i've fixed this with the new commit. let me know if i fixed the issue correctly.

tijlleenders reviewed (approved):
Awesome, thanks!

harshalranjhani commented:
hey @tijlleenders , i was playing around with it when i realised that if the goal was a child goal, the state wasn't preserved and in the background, the parent goal title was disappearing, so i fixed that as well with this new commit and now everything should work perfectly fine!

tijlleenders reviewed (dismissed):
Good catch!

vinaybadgujar102 commented on src/components/GoalsComponents/GoalSublist/components/GoalHistory.tsx:
```diff
@@ -24,6 +24,7 @@ const breadcrumbStyle: React.CSSProperties = {
 };
 
 const BreadcrumbItem = ({ goal }: { goal: ISubGoalHistory }) => {
+  console.log(goal.goalTitle);
```
remove this log

harshalranjhani replied:
removed this as well.

vinaybadgujar102 commented on src/hooks/useGoalSelection.ts:
```diff
@@ -14,6 +14,7 @@ export const useGoalSelection = (goals: GoalItem[]): GoalItem | undefined => {
   const downPress = useKeyPress("ArrowDown");
   const rightPress = useKeyPress("ArrowRight");
   const leftPress = useKeyPress("ArrowLeft");
+  const spacePress = useKeyPress(" ");
```
can we use "Space" key code to detect the space key press?

harshalranjhani replied:
@vinaybadgujar102 fixed it.

vinaybadgujar102 commented:
Thanks for your contribution and effort 🙌
